portfolio/src/main/java/com/google/sps/data/TimeRangeGroup.java

```diff
@@ -0,0 +1,97 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** Models a collection of Time Ranges with methods for add and delete. */
+public class TimeRangeGroup {
+
+  public List<TimeRange> allTimeRanges;
+
+  public TimeRangeGroup() {
+    allTimeRanges = new ArrayList();
+  }
+
+  /** Adds a new time range to the list of all time ranges. */
+  public void addTimeRange(TimeRange newTimeRange) {
+    if (!allTimeRanges.contains(newTimeRange)) {
+      allTimeRanges.add(newTimeRange);
+    }
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    return other instanceof TimeRangeGroup && this.equals((TimeRangeGroup) other);
+  }
+
+  /** Returns the list of all time ranges. */
+  public List<TimeRange> getAllTimeRanges() {
+    return allTimeRanges;
+  }
+
+  /**
+   * Deletes an entire time range from the list of all time ranges.
+   *
+   * @return the modifed list of time ranges.
+   */
+  public List<TimeRange> deleteEntireTimeRange(TimeRange timeRange) {
+    if (timeRange == null) {
+      throw new IllegalArgumentException("Time range to delete cannot be null");
+    }
+
+    if (allTimeRanges.contains(timeRange)) {
+      allTimeRanges.remove(timeRange);
+    } else {
+      throw new IllegalArgumentException(
+          "The time range to delete does not exist in the collection of all time ranges.");
+    }
+    return allTimeRanges;
+  }
+
+  /**
+   * Deletes only part of a time range from the middle.
+   *
+   * @return the sorted all time ranges with the new ranges included.
+   */
+  public List<TimeRange> deletePartOfTimeRange(
+      TimeRange originalTimeRange, TimeRange timeRangeToDelete) {
```

Comment:
These variable names are seem inconsistent. I think either one of the pairs `originalTimeRange, toDeleteTimeRange` or `timeRangeOriginal, timeRangeToDelete` would be more consistent.

Reply:
Updated.